src/components/CustomDropdown/index.tsx

```diff
@@ -29,17 +32,65 @@ interface LabelValues {
     imgUrl: string;
 }
 
-export default function CustomDropdown(props: CustomDropdownContainerProps): ReactElement {
-    const [options, setOptions] = useState<Option[]>([]);
-    const [value, setValue] = useState<Option>();
+interface State {
+    value: Option;
+}
+
+const pageSize = 10;
+
+const CreatablePaginate = withAsyncPaginate(Creatable);
+const SelectPaginate = withAsyncPaginate(Select);
+
+export default class CustomDropdown extends Component<CustomDropdownContainerProps, State> {
```

Comment:
Why refactor from function component with hooks to a class component with state?

Reply:
The only reason was that we need `UNSAFE_componentWillReceiveNewProps`. With hooks we can't catch props update before component re-renders. Check `loadOptions` function which awaits for new props to arrive.